rust/src/applayertemplate/logger.rs

```diff
@@ -0,0 +1,41 @@
+/* Copyright (C) 2018 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+use libc;
+use std;
+use json::*;
+use super::TemplateTransaction;
```

Comment:
what does this do?

Reply:
This pulls TemplateTransaction into scope from mod.rs.
 
Something more general...  We have been doing stuff like "dhcp/mod.rs", "dhcp/dhcp.rs". It seems idiomatic in Rust to put the stuff that you might put into a file named the same as the module, into mod.rs itself. So I went with that approach there. Even if I had used templater.rs, we'd have something like 'super::template::TemplateTransaction' or '::applayertemplate::template::TemplateTransaction' where "super" is the preferred way.

Reply:
Hmm not sure I'm a fan on this. I actually like the separation of mod.rs and actual content in other parsers. Anyway we can defer this to later.